charms/reactive/relations.py

```diff
@@ -200,10 +203,14 @@ def _find_impl(cls, role, interface):
         """
         # The module has already been discovered and imported.
         module = 'relations.{}.{}'.format(interface, role)
-        if module in sys.modules:
-            return cls._find_subclass(sys.modules[module])
-        else:
-            return None
+        if module not in sys.modules:
+            try:
+                _append_path(hookenv.charm_dir())
+                _append_path(os.path.join(hookenv.charm_dir(), 'hooks'))
```

Comment:
I suppose it would be better to have a helper that does these two _append_path calls, shared here and with discover() in bus.py. But that means more lines, and its highly unlikely we will be adding more cruft to sys.path in the future.